vmdk_plugin/utils/refcount/refcnt.go

```diff
@@ -150,10 +176,49 @@ func newRefCount() *refCount {
 	}
 }
 
-// Init Refcounts. Discover volume usage refcounts from Docker.
-// This functions does not sync with mount/unmount handlers and should be called
-// and completed BEFORE we start accepting Mount/unmount requests.
-func (r RefCountsMap) Init(d drivers.VolumeDriver, mountDir string, name string) {
+// return if refcounting has been successfull
+func (r *RefCountsMap) GetSuccess() bool {
+	return r.refcntSuccess
+}
+
+// dirty the background refcount process
+func (r *RefCountsMap) Dirty() {
+	r.curr_state.mtx.Lock()
+	r.curr_state.dirty = true
+	r.curr_state.mtx.Unlock()
+}
+
+// tries to calculate refCounts for dvs volumes. If failed, triggers a timer
+// based reattempt to schedule scan after a delay
+func (r *RefCountsMap) Init(d drivers.VolumeDriver, mountDir string, name string) {
+	err := r.calculate(d, mountDir, name)
+	// If refcounting wasn't successfull, schedule one again
+	if err != nil {
+		log.Infof("Refcounting failed: (%v).", err)
+		r.retryCalculate(d, mountDir, name, refCountRetryAttempts-1)
+	}
+}
+
+// create a timer to calculate refcount after a delay. If failed, retry again
+// until retry attempt limit reached
+func (r *RefCountsMap) retryCalculate(d drivers.VolumeDriver, mountDir string, name string, attemptLeft int) {
+	// create a random delay between max and min delay
+	delay := rand.Intn(refCountDelayMax - refCountDelayMin + 1) + refCountDelayMin
+	log.Infof("Scheduling again after %s seconds", delay)
+	timer := time.NewTimer(time.Duration(delay) * time.Second)
+
+	go func() {
+		<-timer.C
+		err := r.calculate(d, mountDir, driverName)
+		if err != nil && attemptLeft > 0 {
```

Comment:
We should exit if we couldn't initialize refCnts in refCountRetryAttempts. If we let plugin continue then disks will never be detached.